cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -5039,6 +5054,154 @@ if(ARROW_S3)
+                      DEPENDS azure_sdk_dependencies)
+  add_library(Azure::azure-core STATIC IMPORTED)
+  set_target_properties(Azure::azure-core
+                        PROPERTIES IMPORTED_LOCATION ${AZURE_CORE_STATIC_LIBRARY}
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   ${AZURESDK_INCLUDE_DIR})
+  set_property(TARGET Azure::azure-core
+               PROPERTY INTERFACE_LINK_LIBRARIES CURL::libcurl OpenSSL::Crypto OpenSSL::SSL)
+  add_dependencies(Azure::azure-core azuresdk_ep)
+
+  add_library(Azure::azure-identity STATIC IMPORTED)
+  set_target_properties(Azure::azure-identity 
+                        PROPERTIES IMPORTED_LOCATION ${AZURE_IDENTITY_STATIC_LIBRARY}
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   ${AZURESDK_INCLUDE_DIR})
+  set_property(TARGET Azure::azure-identity
+               PROPERTY INTERFACE_LINK_LIBRARIES OpenSSL::Crypto Azure::azure-core)
+  add_dependencies(Azure::azure-identity azuresdk_ep)
+
+  add_library(Azure::azure-storage-common STATIC IMPORTED)
+  set_target_properties(Azure::azure-storage-common
+                        PROPERTIES IMPORTED_LOCATION ${AZURE_STORAGE_COMMON_STATIC_LIBRARY}
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   ${AZURESDK_INCLUDE_DIR})
+  set_property(TARGET Azure::azure-storage-common
+               PROPERTY INTERFACE_LINK_LIBRARIES OpenSSL::Crypto LibXml2::LibXml2 Azure::azure-core)
+  add_dependencies(Azure::azure-storage-common azuresdk_ep)
+
+  add_library(Azure::azure-storage-blobs STATIC IMPORTED)
+  set_target_properties(Azure::azure-storage-blobs
+                        PROPERTIES IMPORTED_LOCATION ${AZURE_STORAGE_BLOBS_STATIC_LIBRARY}
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   ${AZURESDK_INCLUDE_DIR})
+  set_property(TARGET Azure::azure-identity
+               PROPERTY INTERFACE_LINK_LIBRARIES Azure::azure-core Azure::azure-storage-common)
+  add_dependencies(Azure::azure-storage-blobs azuresdk_ep)
+
+  add_library(Azure::azure-storage-files-datalake STATIC IMPORTED)
+  set_target_properties(Azure::azure-storage-files-datalake 
+                        PROPERTIES IMPORTED_LOCATION ${AZURE_STORAGE_FILES_DATALAKE_STATIC_LIBRARY}
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   ${AZURESDK_INCLUDE_DIR})
+  set_property(TARGET Azure::azure-storage-files-datalake
+               PROPERTY INTERFACE_LINK_LIBRARIES Azure::azure-core Azure::azure-identity Azure::azure-storage-common Azure::azure-storage-blobs)
+  add_dependencies(Azure::azure-storage-files-datalake azuresdk_ep)
+
+  set(AZURE_SDK_VENDORED TRUE)
+  set(AZURESDK_LIBRARIES)
+  list(APPEND
+       AZURESDK_LIBRARIES
+       Azure::azure-core
+       Azure::azure-identity
+       Azure::azure-storage-blobs
+       Azure::azure-storage-common
```

Comment:
```suggestion
 ```

Reply:
I tried this but it actually broke the build 
```
/usr/bin/ld: debug/libarrow.so.1300.0.0: undefined reference to `Azure::Identity::ManagedIdentityCredential::ManagedIdentityCredential(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Azure::Core::Credentials::TokenCredentialOptions const&)'
/usr/bin/ld: debug/libarrow.so.1300.0.0: undefined reference to `Azure::Identity::ManagedIdentityCredential::~ManagedIdentityCredential()'
```

Reply:
Oh. I'll take a look at this.

Reply:
I think this is superseded by the switch to using `FetchContent` https://github.com/apache/arrow/pull/36835#discussion_r1298073330